clang/test/dpct/helper_files_ref/include/lapack_utils.hpp

```diff
@@ -353,6 +354,689 @@ inline int potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, int n,
   return 0;
 #endif
 }
+
+namespace detail {
+template <template <typename> typename functor_t, typename... args_t>
+inline int lapack_shim(sycl::queue &q, library_data_t a_type, int *info,
+                       std::string const &lapack_api_name, args_t... args) {
+  auto handle_lapack_exception = [&](const oneapi::mkl::lapack::exception &e) {
+    std::cerr << "Unexpected exception caught during call to LAPACK API: "
+              << lapack_api_name << std::endl
+              << "reason: " << e.what() << std::endl
+              << "info: " << e.info() << std::endl
+              << "detail: " << e.detail() << std::endl;
+    int info_val = static_cast<int>(e.info());
```

Comment:
e.info() returns int64_t, and truncating it to int may wrongly make info_val zero. Suggestion is to use int64_t or auto for info_val declaration and remove static_cast.

Reply:
I think there is no way to avoid the truncating. Because in the original code (CUDA), user may only allocated sizeof(int) memory. If I use int64_t here, there will be an overflow in user's code.
I will throw an exception here if e.info() exceed the range of int.

Reply:
Ok, sounds good. Thanks for the answers and updates!